main/solution/backend/src/lambdas/aws-account-onboarding/handler.js

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const ServicesContainer = require('@aws-ee/base-services-container/lib/services-container');
+const { registerServices } = require('@aws-ee/base-services/lib/utils/services-registration-util');
+const { getSystemRequestContext } = require('@aws-ee/base-services/lib/helpers/system-context');
+
+const pluginRegistry = require('./plugins/plugin-registry');
+
+const handler = async () => {
+  const container = new ServicesContainer(['settings', 'log']);
+  // registerServices - Registers services by calling each service registration plugin in order.
+  await registerServices(container, pluginRegistry);
+  await container.initServices();
+  await handlerWithContainer(container);
+};
+
+const handlerWithContainer = async container => {
+  const awsCfnService = await container.find('awsCfnService');
+  const userContext = getSystemRequestContext();
+
+  // Check permission status of all accounts
+  await awsCfnService.batchCheckAccountPermissions(userContext);
```

Comment:
Emmmm should the batchCheckAccountPermissions method calls attemptOnboardAccounts method? What's the benefit of exposing both methods?

Reply:
it isn't, this is just the lambda function that's running in the backend. I set up the lambda to automatically refresh the permissions status and try to onboard accounts, so that neither method has to be linked to the heartbeat function. I can probably also take the permissionsCheck out of the heartbeat call in the frontend piece, but I thought it was better to get the faster update when the user is actually on the aws accounts page.

Reply:
just refactored this 👍